src/fzx/tui/color.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "fzx/macros.hpp"
+#include <cstdint>
+#include <sstream>
+#include <fmt/core.h>
+struct Color
+{
+  uint8_t mRed, mGreen, mBlue;
```

Comment:
There also has to be some way of representing predefined 8/16 terminal colors + default color. Maybe an additional byte that encodes the color type?

Reply:
rename existing one to `Truecolor`, then `typedef Color = std::variant<Truecolor, Eight, Sixteen>`?

Reply:
Sure, either std::variant or something like this
```cpp
struct Color
{
  enum class Type : uint8_t { Term, RGB };

  Color() = default;

  /// Construct 16 colors
  static Color term(uint8_t color) {
    Color c;
    c.mType = Type::Term;
    c.mColors[0] = color;
    return c;
  }

  /// Construct RGB
  static Color rgb(uint8_t r, uint8_t g, uint8_t b) {
    Color c;
    c.mType = Type::RGB;
    c.mColors[0] = r;
    c.mColors[1] = g;
    c.mColors[2] = b;
    return c;
  }

  Type type() const { return mType; }
  uint8_t term() const { return mColors[0]; }
  uint8_t red() const { return mColors[0]; }
  uint8_t green() const { return mColors[1]; }
  uint8_t blue() const { return mColors[2]; }

private:
  Type mType { Type::Term };
  uint8_t mColors[3] {};
};
```

Reply:
```cpp
/// Construct 16 colors
static constexpr Color term(uint8_t color) {
    Color c;
    c.mType = Type::Term;
    c.mColors[0] = color;
    return c;
}
```
what does this mean?

Reply:
Wasn't sure how to name the variant that holds the 16 colors, vim uses `cterm` for it.

It should correspond to `\e[3Nm` (dark, 0-7 range) and `\e[9Nm` (light, 0-7) for the foreground, and `\e[4Nm` (dark, 0-7) and `\e[10Nm` (light, 0-7) for background. `\e[39m` and `\e[49m` are the default background and foreground

Reply:
does "dark" and "light" mean "normal" and "bright" here? (take a look at the latest commit)

Reply:
Yes

Reply:
done

Reply:
my std::variant version is 4 bytes, same as yours